transitions/generate_patient_id_on_patients.js

```diff
@@ -0,0 +1,11 @@
+var transitionUtils = require('./utils');
+
+module.exports = {
+  filter: function(doc) {
+    return doc.type === 'person' &&
```

Comment:
(having said all that, a reasonably quick fix would be, since ID generation is currently centralised, would be to call the view on startup and then keep the results in memory. That way you don't need use IO to verify collisions. You could even generate N in advance and then just hand them out, if you wanted to split up the time to generate from sentinel walking through the changes feed )